recipes/arrow/all/conanfile.py

```diff
@@ -351,6 +360,7 @@ def generate(self):
         tc.variables["ARROW_WITH_THRIFT"] = bool(self.options.with_thrift)
         tc.variables["Thrift_SOURCE"] = "SYSTEM"
         if self.options.with_thrift:
+            tc.variables["ARROW_THRIFT"] = True
```

Comment:
As far as I can see, this variable was not set, but maybe we would need to set `ARROW_WITH_THRIFT` instead? Am I missing something? Do you have any insight? Thanks!

Reply:
I am not super familiar with Arrow's code base, so I am not 100% sure on what's the difference between `ARROW_WITH_THRIFT` and `ARROW_THRIFT`.

A quick search on Arrow's codebase suggests that `ARROW_THRIFT` is only referenced in

```console
$ find arrow -type f -exec grep -P "\bARROW_THRIFT\b" {} +

arrow/cpp/cmake_modules/ThirdpartyToolchain.cmake:if(ARROW_THRIFT)
arrow/python/build/lib.linux-x86_64-cpython-312/cmake_modules/ThirdpartyToolchain.cmake:if(ARROW_THRIFT)
```

Where it seems to be used only to turn on ZLIB support  - [link](https://github.com/apache/arrow/blob/817ccbcb05e958d67927a998cc186bfed0d8528f/cpp/cmake_modules/ThirdpartyToolchain.cmake#L389-L392)

```cmake
if(ARROW_THRIFT)
  set(ARROW_WITH_ZLIB ON)
endif()
```

So I think setting both `ARROW_WITH_THRIFT` and `ARROW_THRIFT` is ok (will push a commit to do so ASAP).

Reply:
Thanks!